addons/mail/models/mail_blacklist.py

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class MailBlackList(models.Model):
+    """ Model of blacklisted email addresses to stop sending emails."""
+    _name = 'mail.blacklist'
+    _inherit = ['mail.thread']
+    _description = 'Mail Blacklist'
+    _rec_name = 'email'
+
+    email = fields.Char(string='Email Address', required=True, index=True, help='This field is case insensitive.',
+                        track_visibility='always')
+    active = fields.Boolean(default=True, track_visibility='onchange')
+
+    _sql_constraints = [
+        ('unique_email', 'unique (email)', 'Email address already exists!')
+    ]
+
+    @api.model
+    def create(self, values):
+        """ To avoid crash during import due to unique email, return the existing record if any """
+        rec = self.search([('email', '=ilike', values['email'])])
+        if len(rec) == 1:
+            return rec
+        return super(MailBlackList, self).create(values)
+
+    def _add(self, email):
```

Comment:
`@api.model` ?

Reply:
Not necessary on private methods to say model / multi. ORM is smart enough to handle it properly :).